internal/v1/handler_compose_image.go

```diff
@@ -1256,5 +1277,61 @@ func (h *Handlers) buildCustomizations(ctx echo.Context, cr *ComposeRequest, d *
 		}
 	}
 
+	if cust.Aap != nil {
+		data := common.ToPtr(fmt.Sprintf(
+			"#!/usr/bin/bash\ncurl -s -i --data %q %s",
+			fmt.Sprintf("host_config_key=%s", cust.Aap.HostConfigKey),
+			fmt.Sprintf(
+				"%s/api/v2/job_templates/%v/callback/",
+				cust.Aap.AnsibleControllerUrl,
+				cust.Aap.JobTemplateId,
+			),
+		))
+
+		files := []composer.File{
+			// This follows the convention used in the frontend here:
+			// https://github.com/osbuild/image-builder-frontend/blob/e500efc56bd8fd72ba920825f68b031067edd8c8/src/Components/CreateImageWizard/utilities/requestMapper.ts#L566-L577
+			// This is not done in the frontend though, so we have the ability to use
+			// make AAP requests using the image-builder-api
+			{
+				Data:          data,
+				Path:          fmt.Sprintf("/usr/local/sbin/%s", AAP_FIRST_BOOT_FILENAME),
+				Mode:          common.ToPtr("0774"),
+				EnsureParents: common.ToPtr(true),
+			},
+			{
+				Data:          common.ToPtr(AAP_FIRST_BOOT_SERVICE),
+				Path:          fmt.Sprintf("/etc/systemd/system/%s.service", AAP_FIRST_BOOT_FILENAME),
+				EnsureParents: common.ToPtr(true),
+			},
+		}
+
+		if cust.Aap.TlsCertificate != nil {
+			if res.Cacerts != nil && res.Cacerts.PemCerts != nil {
+				res.Cacerts.PemCerts = append(res.Cacerts.PemCerts, *cust.Aap.TlsCertificate)
+			} else {
+				res.Cacerts = &composer.CACertsCustomization{
+					PemCerts: []string{*cust.Aap.TlsCertificate},
+				}
+			}
+		}
+
+		if res.Services != nil && len(*res.Services.Enabled) > 0 {
+			if res.Services.Enabled != nil && len(*res.Services.Enabled) > 0 {
+				res.Services.Enabled = common.ToPtr(append(*res.Services.Enabled, fmt.Sprintf("%s.service", AAP_FIRST_BOOT_FILENAME)))
+			}
+		} else {
+			res.Services = &composer.Services{
+				Enabled: &[]string{fmt.Sprintf("%s.service", AAP_FIRST_BOOT_FILENAME)},
+			}
+		}
```

Comment:
The build-in function `append` does work fine with `nil` slice, you can append to `nil` slice and it will allocate it automatically. The code can be much more simple in here.

Reply:
Ah thanks, yeah this was pretty ugly. I actually didn't know you could append to a nil slice. Will sort that out